da/da.go

```diff
@@ -90,43 +91,41 @@ func (d *DASubmitMetaData) ToPath() string {
 // FromPath parses a path to a DAMetaData.
 func (d *DASubmitMetaData) FromPath(path string) (*DASubmitMetaData, error) {
 	pathParts := strings.FieldsFunc(path, func(r rune) bool { return r == '.' })
+	if len(pathParts) < 2 {
+		return nil, fmt.Errorf("invalid DA path")
+	}
+
 	height, err := strconv.ParseUint(pathParts[1], 10, 64)
 	if err != nil {
 		return nil, err
 	}
-	if len(pathParts) > 2 {
-		index, err := strconv.Atoi(pathParts[2])
+
+	submitData := &DASubmitMetaData{
+		Height: height,
+		Client: Client(pathParts[0]),
+	}
+
+	if len(pathParts) > 7 {
```

Comment:
i guess the if should compare len(pathParts) > 2 instead of > 7

Reply:
why that? if u access `pathParts[6]` u need to validate it exists

Reply:
but if the length of pathParts is 7, that code will never by executed with the current if statement. shouldnt it be len(pathPars)== 7?